tools/memap.py

```diff
@@ -486,13 +495,13 @@ def main():
 
     parser.add_argument('file', help='memory map file')
 
-    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (ARM, GCC_ARM, IAR)',\
-                        required=True)
+    parser.add_argument('-t', '--toolchain', dest='toolchain', help='select a toolchain used to build the memory map file (%s)' % ", ".join(MemapParser.toolchains),\
```

Comment:
> The only issue with this is for the ARM toolchain. In all of the CLI scripts, ARM==ARM_STD and uARM==ARM_MICRO. I'm not familiar with this script and what exactly it expects, perhaps @mlnx can elaborate a bit?

> But if this script is to be invoked by the user, than they should be using `ARM` and `uARM`, not `ARM_STD` and `ARM_MICRO`.

The script doesn't really care if you compile with ARM or uARM (the memory map file format is the same, and should produce the same information). I think it would be good to use the same toolchain names as `mbed-cli`, so users do not get confused.

I understand the concern on having the toolchain names in two places, but I guess we may need to do that if we want to let users invoke the utility manually. There might be a better way to solve this.